src/main/java/com/google/idea/perf/Matching.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.idea.perf
+
+class MatchResult(
+    val source: String,
+    val matchedChars: List<Int>,
+    val score: Int
+)
+
+fun fuzzyMatchMany(sources: Collection<String>, pattern: String): List<MatchResult> {
+    val matches = ArrayList<MatchResult>(sources.size)
+
+    for (source in sources) {
+        matches.add(fuzzyMatch(source, pattern))
+    }
+
+    matches.sortByDescending { it.score }
+    return matches
+}
+
+const val ROOT: Byte = 0
+const val LEFT: Byte = 1
+const val UP: Byte = 2
+const val DIAGONAL: Byte = 3
+
+const val MATCH_SCORE = 4
+const val MISMATCH_SCORE = -4
+const val GAP_SCORE = -1
+
+fun fuzzyMatch(source: String, pattern: String): MatchResult {
```

Comment:
It might be nice to add a comment providing the name of the algorithm, and/or a link to any websites helpful for understanding it.

Reply:
Sure!